### ISSUE TITLE
Type variable of generic type is not concretised when parameters captured with ParamSpec

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1307]
**Describe the bug**
Callables annotated with `ParamSpec` do not carry over concrete parameters of generic types.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of `foo(baz.qux)` would be `(v: int) -> None`.

**Screenshots or Code**
```python
from typing import Callable, Generic, TypeVar
from typing_extensions import ParamSpec

T = TypeVar("T")
P = ParamSpec("P")
C = TypeVar("C", bound=Callable[..., object])

def foo(f: Callable[P, T]) -> Callable[P, T]:
    ...

def bar(f: C) -> C:
    ...

class Baz(Generic[T]):
    def qux(self, v: T) -> None:
        ...


baz: Baz[int] = Baz()
# Type of "baz.qux" is "(v: int) -> None"
reveal_type(baz.qux)
# Type of "foo(baz.qux)" is "(v: T) -> None"
reveal_type(foo(baz.qux))
# Type of "bar(baz.qux)" is "(v: int) -> None"
reveal_type(bar(baz.qux))

```

**VS Code extension or command-line**
Either, 1.1.96

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.97, which I just published. It will also be included in the next release of pylance.
